airflow/www/forms.py

```diff
@@ -204,6 +204,12 @@ def _iter_connection_types() -> Iterator[tuple[str, str]]:
                 yield (connection_type, provider_info.hook_name)
 
     class ConnectionForm(DynamicForm):
+        def process(self, formdata=None, obj=None, **kwargs):
+            super().process(formdata=formdata, obj=obj, **kwargs)
+            for field in self._fields.values():
+                if hasattr(field, "data") and isinstance(field.data, str):
```

Comment:
```suggestion
                if isinstance(getattr(field, "data", None), str):
```

Reply:
Good suggestion. Optimises the code readability.